### PR TITLE
Dnsmasq

### PR DESCRIPTION
**Pull request by HenrithicusGreenson**

### CODE REVIEW

File: genisys/modules/dns.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    DNS_FILE = 'dnsmasq.conf'
+    def __init__(self: Self, config):
+        """Temporarily seeing if this portion of the config file is applicable to DNSMasq"""
+        self.config = config.getSection("DNSMasq Overrides")
+    def install_location(self: Self) -> Path:
+        """This is where the DNSMasq config file is stored/accessed"""
+        return Path(DNS_DIR, DNS_FILE)
+    def generate(self: Self) -> str:
+        #consider adding DNSStubListener=no to /etc/systemd/resolved.conf to prevent port 53 errors?
+        backup = Path(DNS_DIR, 'dnsmasq.conf.orig')
```
Comment (gingrm02) [inline genisys/modules/dns.py:17]: Move to Base.py

File: genisys/modules/dns.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        self.config["network"] = config.getSection("Network")
+        #Adding override config to read the last field in our config file
+        self.config["overrides"] = config.getSection("DNSMasq Overrides")
+    def install_location(self: Self) -> Path:
+        """This is where the DNSMasq config file is stored/accessed"""
+        return Path(DNS_DIR, DNS_FILE)
+    def generate(self: Self) -> str:
+        configWriter = ''
+        #The below line deals with the port 53 resolved issue when running dnsmasq after installation
+        configWriter+="bind-interfaces"
```
Comment (gingrm02) [inline genisys/modules/dns.py:20]: Missing newline may cause issues with the file

File: pyproject.toml [hunk 3]
```diff
@@ -23,6 +23,9 @@ genisys = "genisys.main:main"
 
 [tool.poetry.dependencies]
 python = "^3.8"
+<<<<<<< HEAD
+PyYAML = "^5.1"
```
Comment (gingrm02) [inline pyproject.toml:27]: This isn't showing as a merge conflict even though it looks like one? Fix this file with the correct requirements before we can merge

**Review by gingrm02 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (patedm01) [reply pyproject.toml:26]: I'm not sure if the change I pushed fixed the issue. Please look over the file to confirm.

**Review by gingrm02 — APPROVED** [review]

**Review by BBergle — APPROVED** [review]

**Review by r-akers — APPROVED** [review]

**Pull request merged**